### ISSUE TITLE
Spurious MGLOfflineStorage unit test failures on iOS build bot

### ISSUE THREAD

**Issue by jfirebaugh** [mapbox/mapbox-gl-native#6385]
The following tests are spuriously failing iOS builds with regularity:

```
    ✗ testAddPack, Asynchronous wait failed: Exceeded timeout of 1 seconds, with unfulfilled expectations: "add pack completion handler", "Expect change to packs of <MGLOfflineStorage: 0x7fd210cd1340>".
    ✗ testAddPack, (([MGLOfflineStorage sharedOfflineStorage].packs.count) equal to (countOfPacks + 1)) failed: ("0") is not equal to ("1") - Added pack should have been added to the canonical collection of packs owned by the shared offline storage object. This assertion can fail if this test is run before -testAAALoadPacks.
    ✗ testAddPack, ((pack.region) equal to (region)) failed: ("(null)") is not equal to ("<MGLTilePyramidOfflineRegion: 0x7fd213408730>") - Added pack’s region has changed.
    ✗ testAddPack, (([userInfo isKindOfClass:[NSDictionary class]]) is true) failed - Context of offline pack isn’t a dictionary.
    ✗ testAddPack, (([userInfo[nameKey] isKindOfClass:[NSString class]]) is true) failed - Name of offline pack isn’t a string.
    ✗ testAddPack, ((userInfo[nameKey]) equal to (name)) failed: ("(null)") is not equal to ("🍇 Grape Grove") - Name of offline pack has changed.
    ✗ testAddPack, ((pack.state) equal to (MGLOfflinePackStateInactive)) failed: ("0") is not equal to ("1") - New pack should initially have inactive state.
```

**Comment by 1ec5** [comment]
The first test started failing reliably about a month ago, causing all the other tests to cascade, without any relevant SDK-level changes. #6007 improved their reliability, but not completely. Let's try increasing the timeout a bit; this might be indicative of a performance problem.

**Comment by 1ec5** [comment]
This failure only affects some of my branches, including #5999. 61043b5ac67123bd0674808cf6d6133ef6d2a285 is an attempt to close one potential race condition. I'll split that out and merge it separately to see if it helps.

**Comment by 1ec5** [comment]
We’re bumping the timeout in #6397.

**Comment by jfirebaugh** [comment]
Seen again in https://www.bitrise.io/build/236fd3bed24e11e1.

**Comment by 1ec5** [comment]
Seen also in https://www.bitrise.io/build/236fd3bed24e11e1.

I think this may be related to #5996, which for whatever reason only reproduces on macOS.

**Comment by boundsj** [comment]
#5996 changed the `bitrise.yml` file for macOS so that artifacts with additional context are created for each build. The fix for #5996 probably won't help with the spurious failure noted here but, if it happens again, the additional logging may be useful. I'll keep this open and keep 👀. Please continue to post the bitrise build results here if you see a macOS failure especially now that we have additional logging in place.

cc @1ec5 @jfirebaugh

**Comment by boundsj** [comment]
I've not seen any trace of this issue for some time now. I'm going to close for now. Please reopen and paste the failed bitrise build link here if you see this again.
